src/Domain/CreateInitialElement.cpp

```diff
@@ -43,7 +43,7 @@ Element<VolumeDim> create_initial_element(
         // SegmentIds of the current element using the neighbor's axes.
         auto segment_ids_of_unrefined_neighbor = orientation(segment_ids);
         const auto& refinement_of_neighbor =
-            initial_refinement_levels[block_neighbor.id()];
+            initial_refinement_levels[*block_neighbor.begin()];
```

Comment:
Assert that `block_neighbor.size() == 1` here and in other places you assume this?